server/datastore/mysql/queries.go

```diff
@@ -41,131 +41,115 @@ func (ds *Datastore) ApplyQueries(ctx context.Context, authorID uint, queries []
 }
 
 func (ds *Datastore) applyQueriesInTx(ctx context.Context, authorID uint, queries []*fleet.Query) (err error) {
-	tx, err := ds.writer(ctx).BeginTxx(ctx, nil)
-	if err != nil {
-		return ctxerr.Wrap(ctx, err, "begin applyQueriesInTx")
-	}
-
-	defer func() {
-		if err != nil {
-			rbErr := tx.Rollback()
-			// It seems possible that there might be a case in
-			// which the error we are dealing with here was thrown
-			// by the call to tx.Commit(), and the docs suggest
-			// this call would then result in sql.ErrTxDone.
-			if rbErr != nil && rbErr != sql.ErrTxDone {
-				panic(fmt.Sprintf("got err '%s' rolling back after err '%s'", rbErr, err))
+	err = ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
```

Comment:
Replace previous `BeginTxx` with `ds.withRetryTxx`.  These both ultimately operate on the writer, but the latter returns a `sqlx.ExtContext` rather than a `sqlx.Tx`, which is what `updateQueryLabelsInTx` was expecting, requiring some other updates.